react-components/src/components/NodeCacheProvider/FdmNodeCache.ts

```diff
@@ -28,36 +31,139 @@ import { partition } from 'lodash';
+    );
+
+    const nonCachedModelMappings = await this.getNonCachedModelMappings(
+      externalIds,
+      nonCachedModelRevisionIds
+    );
+
+    const cachedModelMappings = this.getCachedModelMappings(cachedModelRevisionIds, externalIds);
+
+    const combinedList = [...cachedModelMappings, ...nonCachedModelMappings];
+    return combinedList;
+  }
+
+  private getCachedModelMappings(
+    modelRevisionIds: ModelRevisionId[],
+    externalIds: DmsUniqueIdentifier[]
+  ): ThreeDModelMappings[] {
+    const inputExternalIdSet = new Set<FdmKey>(
+      externalIds.map((id) => createFdmKey(id.space, id.externalId))
+    );
+
+    return modelRevisionIds.map((modelRevisionId) => {
+      return this.getCachedModelMappingForRevision(modelRevisionId, inputExternalIdSet);
+    });
+  }
+
+  private getCachedModelMappingForRevision(
+    modelRevisionId: ModelRevisionId,
+    relevantFdmKeySet: Set<FdmKey>
+  ): ThreeDModelMappings {
+    const revisionCache = this.getOrCreateRevisionCache(
+      modelRevisionId.modelId,
+      modelRevisionId.revisionId
+    );
+
+    const relevantCachedEdgeData = intersectWithStartNodeIdSet(
+      revisionCache.getAllEdges(),
+      relevantFdmKeySet
+    );
+
+    const mappings = new Map(
+      relevantCachedEdgeData.map((data) => [data.edge.startNode.externalId, data.node])
+    );
+
+    return {
+      ...modelRevisionId,
+      mappings
+    };
+  }
+
+  private async getNonCachedModelMappings(
+    uniqueIds: DmsUniqueIdentifier[],
+    modelRevisions: ModelRevisionId[]
+  ): Promise<ThreeDModelMappings[]> {
+    if (modelRevisions.length === 0) {
+      return [];
+    }
+
+    const fdmKeySet = new Set(uniqueIds.map((id) => createFdmKey(id.space, id.externalId)));
+
+    const revisionToEdgesMap = await this.getAndCacheRevisionToEdgesMap(modelRevisions);
+
+    const modelDataPromises = modelRevisions.map(async ({ modelId, revisionId }) => {
+      const revisionKey = createModelRevisionKey(modelId, revisionId);
+      const edges = revisionToEdgesMap.get(revisionKey);
+
+      return this.getRelevantExternalIdToNodeMapForRevision(
+        { modelId, revisionId },
+        edges,
+        fdmKeySet
+      );
+    });
+
+    return await Promise.all(modelDataPromises);
+  }
+
+  private getRelevantExternalIdToNodeMapForRevision(
+    { modelId, revisionId }: ModelRevisionId,
+    edges: FdmEdgeWithNode[] | undefined,
+    relevantFdmKeySet: Set<FdmKey>
+  ): ThreeDModelMappings {
+    if (edges === undefined || edges.length === 0)
+      return { modelId, revisionId, mappings: new Map<CogniteExternalId, Node3D>() };
+
+    const relevantEdges = intersectWithStartNodeIdSet(edges, relevantFdmKeySet);
+
+    const externalIdToNodeMap = new Map<CogniteExternalId, Node3D>(
+      relevantEdges.map((edge) => [edge.edge.startNode.externalId, edge.node])
+    );
+
+    return {
+      modelId,
+      revisionId,
+      mappings: externalIdToNodeMap
+    };
+  }
+
   public async getAllMappingExternalIds(
```

Comment:
As it is exposed as public  function, what will it return if we pass empty array of `ModelRevisionId`?